lib/Dialect/TritonGPU/Transforms/AMDReorderInstructions.cpp

```diff
@@ -0,0 +1,397 @@
+#include "mlir/Analysis/SliceAnalysis.h"
+#include "mlir/IR/Dominance.h"
+#include "mlir/IR/Verifier.h"
+#include "triton/Dialect/TritonGPU/IR/Dialect.h"
+#include "triton/Dialect/TritonGPU/Transforms/Passes.h"
+
+#define GEN_PASS_CLASSES
+#include "triton/Dialect/TritonGPU/Transforms/Passes.h.inc"
+
+using namespace mlir;
+namespace tt = triton;
+namespace ttg = triton::gpu;
+
+class TritonAMDGPUReorderInstructionsPass
+    : public TritonAMDGPUReorderInstructionsBase<
+          TritonAMDGPUReorderInstructionsPass> {
+public:
+  TritonAMDGPUReorderInstructionsPass() = default;
+
+  void sortOperandsByDominance(OperandRange operands,
+                               SmallVector<Value> &operandsSorted) {
+    ModuleOp m = getOperation();
+    mlir::DominanceInfo dom(m);
+
+    for (auto operand : operands) {
+      operandsSorted.push_back(operand);
+    }
+
+    if (operandsSorted.size() == 1) {
+      return;
+    }
+
+    std::sort(operandsSorted.begin(), operandsSorted.end(),
```

Comment:
I am kind of suspicious about this sort...
std::sort assumes you have fully ordered objects, which is not true for values.